org.eclipse.lsp4e/src/org/eclipse/lsp4e/LanguageServerWrapper.java

```diff
@@ -405,20 +405,26 @@ public synchronized void start() throws IOException {
 				this.languageServer.initialized(new InitializedParams());
 			}).thenRun(() -> {
 				final Map<URI, IDocument> toReconnect = filesToReconnect;
-				initializeFuture.thenRunAsync(() -> {
-					watchProjects();
-					for (Entry<URI, IDocument> fileToReconnect : toReconnect.entrySet()) {
-						try {
-							connect(fileToReconnect.getKey(), fileToReconnect.getValue());
-						} catch (IOException e) {
-							throw new RuntimeException(e);
+				final var initializeFuture = this.initializeFuture;
+				if (initializeFuture != null) {
```

Comment:
Should we log something in case initializeFuture is null? This is an unexpected case.